modules/terminate-workers/iam.tf

```diff
@@ -0,0 +1,103 @@
+# ----------------------------------------------------------------------------
+# Terminate Workers - IAM Resources
+# ----------------------------------------------------------------------------
+
+data "aws_iam_policy_document" "assume_role" {
+  count = var.create ? 1 : 0
+
+  statement {
+    actions = [
+      "sts:AssumeRole",
+    ]
+    effect = "Allow"
+
+    principals {
+      identifiers = ["lambda.amazonaws.com"]
+      type        = "Service"
+    }
+  }
+}
+
+resource "aws_iam_role" "lambda" {
+  count = var.create ? 1 : 0
+
+  name                  = var.name_iam_objects == "" ? var.name : var.name_iam_objects
```

Comment:
question: According to the variable description `name_iam_objects` it sounds to me that the name of this and all other IAM objects should be something like `name = "${var.name_iam_objects]-suffix"`

Reply:
Fixed the IAM resource names to be consistent and use the module's "environment" variable in a290efa4934c4a43c3f14143be6f21d5ba1ef689